### PR TITLE
Add request options for pharmacies and urgent care option va and non-va

### PR DESCRIPTION
**Pull request by brangi**
## Description
User stories
As a veteran, I need to be able to search for an urgent care facility so that I can seek care.
As a veteran, I need to be able to search for an urgent care pharmacy so that I can have my urgent care prescription filled in my community.

issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/4924

## Testing done
Locally

## Acceptance criteria
- [ ]department-of-veterans-affairs/va.gov-team#4924

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/facility-locator/config.js [hunk 1]
```diff
@@ -33,9 +35,64 @@ export const api = {
...
+  page,
+  bounds,
+) => {
+  const filterableLocations = ['health', 'benefits', 'cc_provider'];
+  let facility;
+  let service;
+  let url;
+  switch (locationType) {
+    case 'urgent_care':
+      if (!serviceType || serviceType === 'UrgentCare') {
```
Comment (darkbushido) [inline src/applications/facility-locator/config.js]: This is either a 'NonVAUrgentCare' or ('UrgentCare' or nil) you can do this with an if else, if non else everything else.

**Review by darkbushido — CHANGES_REQUESTED** [review]

File: src/applications/facility-locator/config.js (hunk 1 above)
Comment (brangi) [reply src/applications/facility-locator/config.js]: Yeah, done

**Comment by brangi** [comment]
@darkbushido,  I just updated according to your feedback.

**Review by darkbushido — APPROVED** [review]

**Comment by brangi** [comment]
@department-of-veterans-affairs/frontend-review-group , can you provide feedback or approve this PR?

**Review by cvalarida — APPROVED** [review]
I don't know the application well enough to give an in-depth review, but it seems like a lot of changes without corresponding tests.

My two recommendations:
1. Make sure you have a unit test for all changes
    - Even if the application is otherwise untested, this will prevent it from falling further and further behind in test coverage
    - I find it helpful to write the tests _first_ when possible
2. Split up PRs into smaller chunks
    - It's much easier to review a <=300 LoC change than it is ~700
    - This is especially important if peer reviews are your last line of defense against introducing bugs into production

That said, nothing stands out to me as needing changes. It just makes me nervous that I'm not seeing any tests in here.

**Comment by brangi** [comment]
@cvalarida thanks for your comments and approval.  Actually more than  40% of the code added is mock data and tests. But next time I will make sure changes are smaller on my PRs.

**Comment by cvalarida** [comment]
D'oh! Totally skipped over `src/applications/facility-locator/tests/api-url-parameters.spec.js`! You're right. Sorry about that! :grimacing: 

The JSX could probably use some more thorough testing, but I just completely missed the test file you did add. I was looking for a `.unit.spec.js`.

Could you make another PR to add a `.unit` to that filename? Our [test runner won't grab it](src/applications/facility-locator/tests/api-url-parameters.spec.js ) without that in there.

**Comment by brangi** [comment]
@cvalarida, this https://github.com/department-of-veterans-affairs/vets-website/pull/11653 PR takes care of renaming the file among other fixes thanks!

**Pull request merged**